### PR TITLE
Note CLI flag deprecations in documentation

### PR DESCRIPTION
**Pull request by deivid-rodriguez**
## What was the end-user or developer problem that led to this PR?

Documentation was not very clear regarding the deprecation of CLI flags.

## What is your fix for the problem, implemented in this PR?

This PR tries to improve that.

Closes https://github.com/rubygems/rubygems/issues/3600.

# Tasks:

- [x] Describe the problem / feature
- [ ] Write tests
- [x] Write code to solve the problem
- [ ] Get code review from coworkers / friends

I will abide by the [code of conduct](https://github.com/rubygems/rubygems/blob/master/CODE_OF_CONDUCT.md).

### CODE REVIEW

File: bundler/man/bundle-install.ronn [hunk 1]
```diff
@@ -151,11 +169,16 @@ time `bundle install` is run, use `bundle config` (see bundle-config(1)).
...
+  This option is deprecated in favor of the `with` setting.
+
 * `--without=<list>`:
   A space-separated list of groups referencing gems to skip during installation.
   If a group is given that is in the remembered list of groups given
   to --with, it is removed from that list.
 
+  This option is deprecated in favor of the `without` setting.
+
+
```
Comment (colby-swandale) [inline bundler/man/bundle-install.ronn:181]: Is the extra line space needed? otherwise I would remain consistent and only have the 1 line space.

**Review by colby-swandale — APPROVED** [review]

**Comment by colby-swandale** [comment]
Not related to this PR specifically. It's difficult (probably very difficult if you're not familiar with the bundler repo) to figure out exactly what files were changed by a user and what's auto generated 😭

File: bundler/man/bundle-install.ronn (hunk 1 above)
Comment (deivid-rodriguez) [reply bundler/man/bundle-install.ronn:181]: Ooops, my bad. Let me update this!

**Comment by deivid-rodriguez** [comment]
Yeah! Fixing #3918 should make this easier, but there will only be one human readable version of man pages and one autogenerated.

**Comment by colby-swandale** [comment]
Can we just hide everything being autogenerated by default https://docs.github.com/en/github/administering-a-repository/customizing-how-changed-files-appear-on-github ?

**Comment by deivid-rodriguez** [comment]
That sounds like a good idea to me :+1:.

**Comment by deivid-rodriguez** [comment]
@colby-swandale Should we add your suggestion to this PR, or leave it for a separate one?

**Comment by deivid-rodriguez** [comment]
Since you approved previously and mentioned that it was not related to this PR specifically, I'll merge this. Thanks for having a look!

**Comment by colby-swandale** [comment]
Ah sorry, i'm not paying attention to Github at the moment. I agree to make a separate issue/PR

**Comment by deivid-rodriguez** [comment]
No problem at all!

**Pull request merged**